src/handler.py

```diff
@@ -35,37 +32,43 @@
 
 
 try:
-    connection = 'mysql+pymysql://{user}:{password}@{host}/{db_name}'.format(user=rds_config.db_username, password=rds_config.db_password, host=rds_config.rds_host, db_name=rds_config.db_name)
+    connection = 'mysql+pymysql://{user}:{password}@{host}/{db_name}'.format(user=rds_config.db_username,
+                                                                             password=rds_config.db_password,
+                                                                             host=rds_config.rds_host,
+                                                                             db_name=rds_config.db_name)
     engine = create_engine(connection)
     alchemy_utils.Base.metadata.create_all(engine)
     Session = sessionmaker(bind=engine)
+    logger.info("SUCCESS: Connection to RDS mysql instance succeeded")
 except Exception as e:
     logger.error("ERROR: Unexpected error: Could not connect to MySql instance.\t{}".format(e))
     sys.exit()
 
 
-logger.info("SUCCESS: Connection to RDS mysql instance succeeded")
 def scrape(n_tries, departure_airport, arrival_airport, departure_date):
     for i in range(n_tries):
         try:
             legs = get_legs(departure_airport, arrival_airport, departure_date)
-            if legs == []:
-                raise ValueError("No results")
-            persist_legs(legs, departure_date)
-            return 'Scraping successful'
-        except Exception as e:
+            if legs is []:
+                raise ValueError("No results.")
+            persist_legs(legs)
+            return 'Scrape successful.'
+        except Exception as e:  # TODO: import the TimeoutException class and except it and ValueError explicitly
```

Comment:
Do that then :)

Reply:
What about other exceptions? You want to allow them to raise errors?

Reply:
Probably yeah you'll have to test it and see if any other exceptions bubble up that we do want to retry for. But yeah we probably only want to retry for timeout and valuerror. Anything else we probably want to just give up and let the program crash

Reply:
Cool